packages/format-currency/src/index.ts

```diff
@@ -47,11 +47,20 @@ export function createFormatter(): CurrencyFormatter {
 		code: string,
 		options: CurrencyObjectOptions
 	): Intl.NumberFormat {
-		return getCachedFormatter( {
-			locale: getLocaleToUse( options ),
+		const numberFormatOptions: Intl.NumberFormatOptions = {
+			style: 'currency',
 			currency: code,
-			noDecimals: isNoDecimals( number, options ),
-			signForPositive: options.signForPositive ?? false,
+			...( options.stripZeros &&
+				Number.isInteger( number ) && {
+					maximumFractionDigits: 0,
+					minimumFractionDigits: 0,
+				} ),
+			...( options.signForPositive && { signDisplay: 'exceptZero' } ),
+		};
+
+		return getCachedFormatter( {
+			locale: `${ getLocaleToUse( options ) }-u-nu-latn`,
```

Comment:
If we keep using the string version of the latin locale adjustment, can you leave a comment here about why it's present and the existence of the `numberingSystem` [option](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#numberingsystem)? We should try to replace it eventually and I'm certain that I'll forget if this code lives on in any capacity.

Reply:
I will get to it in the end. Don't worry. The caching right now is nearly identical to numberFormat, so I can easily migrate across. I will completely own this to the end. It's on my list.

Reply:
Added the comment back right above.